paasta_tools/contrib/kill_bad_containers.py

```diff
@@ -0,0 +1,78 @@
+#!/opt/venvs/paasta-tools/bin/python3
+import sys
+
+import iptc
+
+from paasta_tools import iptables
+from paasta_tools.utils import get_docker_client
+
+
+def get_container_from_dport(dport, docker_client):
+    for container in docker_client.containers():
+        try:
+            ports = container['Ports']
+            for port in ports:
+                if "PublicPort" in port:
+                    if port["PublicPort"] == int(dport):
+                        return container
+        except KeyError:
+            print(ports)
+            pass
+
+
+def target_rule_to_dport(rule):
+    try:
+        # (('tcp', (('dport', ('31493',)),)),)
+        return rule.matches[0][1][0][1][0]
+    except IndexError:
+        return None
+
+
+def kill_containers_with_duplicate_iptables_rules(docker_client):
+    chain_name = 'DOCKER'
+    table = iptc.Table(iptc.Table.NAT)
+    chain = iptc.Chain(table, chain_name)
+
+    targets_seen = {}
+    raw_rules_seen = {}
+
+    for iptables_rule in chain.rules:
+        rule = iptables.Rule.from_iptc(iptables_rule)
+        target = rule.target_parameters
+        if target not in targets_seen:
+            targets_seen[target] = rule
+            raw_rules_seen[target] = iptables_rule
+        else:
+            print("This is the second time we've seen a rule with the same target_parameters!")
+            print(rule)
+            dport = target_rule_to_dport(rule)
+            container1 = get_container_from_dport(dport, docker_client)
+            print("The other rule with that target is:")
+            print(targets_seen[target])
+            dport = target_rule_to_dport(rule)
```

Comment:
Should this be targets_seen[rule] instead of rule? otherwise it's the same as L48

Reply:
! You are totally right